tests/js/server/shell/shell-foxx-swagger-spec.js

```diff
@@ -377,19 +387,224 @@ describe('Foxx Swagger', function () {
 
     describe('"parameters"', function () {
       describe('in:body', function () {
-        it('TODO');
+        it('includes default schema', function () {
+          service.router.post('/hello', noop);
+          service.buildRoutes();
+          expect(service.docs.paths).to.have.a.property('/hello')
+            .with.a.deep.property('post.parameters[0]')
+            .that.is.eql({
+              in: 'body',
+              name: 'body',
+              schema: true
+            });
+        });
+
+        it('includes explicit joi schema', function () {
+          service.router.post('/hello', noop)
+          .body(joi.string().required());
+          service.buildRoutes();
+          expect(service.docs.paths).to.have.a.property('/hello')
+            .with.a.deep.property('post.parameters[0]')
+            .that.is.eql({
+              in: 'body',
+              name: 'body',
+              description: undefined,
+              required: true,
+              schema: {type: 'string'}
+            });
+        });
+
+        it('includes explicit JSON schema', function () {
+          service.router.post('/hello', noop)
+          .body({type: 'string'});
+          service.buildRoutes();
+          expect(service.docs.paths).to.have.a.property('/hello')
+            .with.a.deep.property('post.parameters[0]')
+            .that.is.eql({
+              in: 'body',
+              name: 'body',
+              required: true,
+              schema: {type: 'string'}
+            });
+        });
+
+        it('includes explicit optional JSON schema', function () {
```

Comment:
As discussed in Slack, we cann add here `optional: true` tests for JOI as well. Otherwise LGTM

Reply:
FYI: I did not review the modules located in "node_modules" itself :P